compiler/typecheck/TcSplice.hs

```diff
@@ -1751,7 +1751,7 @@ reifyTyVars tvs m_tc = mapM reify_tv tvs'
     -- even if the kind is *, we need to include a kind annotation,
     -- in case a poly-kind would be inferred without the annotation.
     -- See #8953 or test th/T8953
-    reify_tv tv = TH.KindedTV name <$> reifyKind kind
+    reify_tv tv = (\k -> TH.KindedTV name k TH.Specified) <$> reifyKind kind -- TODO: I *think* Specified is right here...
```

Comment:
You'll need to modify `reifyTyVars` to take visibility info to make this decision. And then you may have trouble producing that information (say, from datacons). When you do, that identifies more places where you need to store this info.

Reply:
What is `reifyTyVars` doing? I don't think I understand enough about the innards of TH to grasp what it's up to.

Reply:
Let's discuss this at a future meeting. Too hard to explain in this forum. This is the less important end of the patch, though.